src/supabase/queries/auth.ts

```diff
@@ -57,3 +57,27 @@ export const signOutUser = async () => {
     throw error;
   }
 };
+
+export const emailExists = async (email: string): Promise<boolean> => {
+  const { data } = await supabase.from('users').select().eq('email', email);
+  if (!data || data.length === 0) {
+    return false;
+  } else {
+    return true;
+  }
+};
+
+export const passwordExists = async (
+  email: string,
+  password: string,
+): Promise<boolean> => {
+  const { data } = await supabase.auth.signInWithPassword({
```

Comment:
Here `data` is never null and rather defined like this ```const data: {
    user: User;
    session: Session;
} | {
    user: null;
    session: null;
}``` so let's check for `data.user` instead!